OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.java

```diff
@@ -127,6 +124,65 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+        expandedToolbarDismiss = getActivity().findViewById(R.id.expanded_toolbar_dismiss);
+        expandedCourseTitle = getActivity().findViewById(R.id.course_title);
+        MaterialTextView courseOrg = getActivity().findViewById(R.id.course_organization);
+        MaterialTextView courseExpiryDate = getActivity().findViewById(R.id.course_expiry_date);
+        upgradeBtn = getActivity().findViewById(R.id.layout_upgrade_btn);
+        MaterialButton upgradeBtnText = upgradeBtn.findViewById(R.id.btn_upgrade);
+
+        collapsedToolbarTitle.setText(courseData.getCourse().getName());
+        courseOrg.setText(courseData.getCourse().getOrg());
+        expandedCourseTitle.setText(courseData.getCourse().getName());
+
+        String expiryDate = CourseCardUtils.getFormattedDate(requireContext(), courseData);
+        if (!TextUtils.isEmpty(expiryDate)) {
+            courseExpiryDate.setVisibility(View.VISIBLE);
+            courseExpiryDate.setText(expiryDate);
+        }
+
+        if (environment.getConfig().isCourseSharingEnabled()) {
+            expandedCourseTitle.setMovementMethod(LinkMovementMethod.getInstance());
+            SpannableString spannableString = org.edx.mobile.util.TextUtils.setIconifiedText(
+                    requireContext(),
+                    courseData.getCourse().getName(),
+                    R.drawable.ic_share,
+                    v -> ShareUtils.showCourseShareMenu(requireActivity(), expandedCourseTitle,
+                            courseData, analyticsRegistry, environment)
+            );
+            expandedCourseTitle.setText(spannableString);
+        }
+
+        collapsedToolbarDismiss.setOnClickListener(v -> getActivity().finish());
+        expandedToolbarDismiss.setOnClickListener(v -> getActivity().finish());
+
+        ((ShimmerFrameLayout) upgradeBtn).hideShimmer();
+        upgradeBtnText.setText(R.string.value_prop_course_card_message);
+        upgradeBtn.setVisibility(courseData.getMode().equalsIgnoreCase(EnrollmentMode.AUDIT.toString()) ? View.VISIBLE : View.GONE);
+        upgradeBtnText.setOnClickListener(view1 -> CourseModalDialogFragment.newInstance(
+                        Analytics.Screens.PLS_COURSE_DASHBOARD,
+                        courseData.getCourseId(),
+                        courseData.getCourseSku(),
+                        courseData.getCourse().getName(),
+                        courseData.getCourse().isSelfPaced())
+                .show(getChildFragmentManager(), CourseModalDialogFragment.TAG));
+
+        appbar.addOnOffsetChangedListener((appBarLayout, verticalOffset) -> {
+            int maxScroll = appBarLayout.getTotalScrollRange();
+            float percentage = (float) Math.abs(verticalOffset) / (float) maxScroll;
+            handleToolbarVisibility(collapsedToolbar, expandedToolbar, percentage);
+        });
+        ViewAnimationUtil.startAlphaAnimation(collapsedToolbar, View.INVISIBLE);
+    }
+
```

Comment:
there should be a default behavior of collapsing Toolbar animation, cuz this will increases the performance overhead for a lower-end device like nexus 5.

Reply:
The default behavior requires the Toolbar to be of fixed-sized during collapsed state and at the end of the view during the expanded state, which doesn't align with our new design. That's why we implemented a bit more custom design.